### DISCUSSION TITLE
Compiler + Runtime instead of Runtime only

### DISCUSSION THREAD

**Discussion by laven2** [vuepress/core#57] [Q&A] [answered]
Hi,
How to use vue Compiler + Runtime instead of Runtime only in dev mode ?

**Comment by meteorlxy** [comment, marked as answer]
Use `alias` option.

Example:

https://github.com/meteorlxy/vue-showdown/blob/c5710e96789e8bb64ea2871a03a0f9a583c82d10/docs/.vuepress/config.ts#L96-L100

**Reply by laven2** [reply]
Thank you
But... why methods not visible in template? https://github.com/laven2/vuepress-test/blob/master/components/TestElement.vue
only from setup implementation worked

**Reply by meteorlxy** [reply]
The options API is disabled by default. You'll need to enable it via `define` option:

https://github.com/vuepress/vuepress-next/blob/8ba3968a436801e91c306b4791ab14d1b8075dd0/packages/%40vuepress/bundler-webpack/src/config/handlePluginDefine.ts#L24-L26

**Reply by laven2** [reply]
I added this property to define(as define propert and as chainWebpack), but nothing happened
https://github.com/laven2/vuepress-test/blob/f3453d2e97b4889867e6886d25805d62f4850a7b/docs/.vuepress/config.js#L14-L24

In the console I see that `__VUE_OPTIONS_API__ = 'true'` but methods (options API) not visible in template
I get an error `$options.[myMethodName] is not a function`

**Reply by meteorlxy** [reply]
Well, the snippet above is for explaining the reason.

Using `define` options is simpler:

```js
module.exports = {
  define: {
    __VUE_OPTIONS_API__: true,
  },
}
```

**Reply by meteorlxy** [reply]
We may enbale options API by default, but I'm not sure if users prefer using options API or not. That depends on the future feedback.

**Reply by laven2** [reply]
> Well, the snippet above is for explaining the reason.
> 
> Using `define` options is simpler:
> 
> ```js
> module.exports = {
>   define: {
>     __VUE_OPTIONS_API__: true,
>   },
> }
> ```

Thank you, but with this case options API still doesn't work

**Reply by laven2** [reply]
> We may enbale options API by default, but I'm not sure if users prefer using options API or not. That depends on the future feedback.

For me options API will help to quickly migrate from vue 2 to 3 version and support 3rd-party libraries integrations

**Reply by meteorlxy** [reply]
Now options API is enabled by default

**Reply by laven2** [reply]
Great! Thank you! It works
